src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php

```diff
@@ -85,7 +85,7 @@ public static function createConnection($dsn, array $options = [])
                 preg_match($dsnPattern, $server, $matches);
 
                 $username = $matches['username'] ?: $username;
-                $password = $matches['password'] ?: $password;
+                $password = rawurldecode($matches['password'] ?: $password);
```

Comment:
The issue in this class (and the other) is not only about the password - it's about all URL components.
The fix should involve using parse_url (as done in all other DSN parsers) to let PHP decode everything as needed.

Reply:
I updated the DSN parsing with `parse_url` :+1: I'll try to have a look at other adapters in a separate PR :+1: